### PR TITLE
Engaging Crowds: Subject set selection fixes

### LINKED ISSUE #1998: Engaging Crowds: updating the wording on the workflow selection menu
## Package
app-project

## Is your feature request related to a problem? Please describe.
The workflow selection menu doesn't indicate whether each choice acts a button, opening the subject set selection tools (grouped workflows), or a link to the workflow's Classify page (all other workflows.)

Each choice should have a little bit of extra text added to it, indicating what to expect when the button is clicked.
Design for the workflow menu, where each workflow has the words 'set selection' or 'subject selection' added after the workflow name.

### PR DESCRIPTION
**Pull request by eatyourgreens**
This PR reorganises the code for workflow selection on the Classify page, including some small fixes for bugs.
- The popup workflow selector is moved into its own `WorkflowMenu` component, with tests and stories. (@srallen the code here shows an example of documenting functional component props with CSF.)
- the close button should be removed from the workflow popup now.
- the wording on the subject set selector should now say 'Back to workflow choice.'
- workflow buttons should now say 'set selection.'

Closes #1998.

Screenshot of the workflow menu, with no close button and improved wording.

Screenshot of the subject set selector, with no close button and improved wording.

Package:
app-project

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Self-documenting components! The descriptions and default values here are all taken from the `WorkflowMenu` code.
Docs page for the WorkflowMenu component, showing the component description and props list with descriptions and default values.

File: packages/app-project/src/screens/ClassifyPage/components/WorkflowMenu/WorkflowMenu.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+const workflowType = shape({
+  displayName: string,
+  id: string
+})
+
+WorkflowMenu.propTypes = {
+  /**
+    Background colour of the title bar.
+  */
+  headingBackground: string,
```
Comment (shaunanoordin) [inline packages/app-project/src/screens/ClassifyPage/components/WorkflowMenu/WorkflowMenu.js:67]: Wait, we have auto-documentation? I approve of these style of comments!

Comment (eatyourgreens) [reply packages/app-project/src/screens/ClassifyPage/components/WorkflowMenu/WorkflowMenu.js:67]: I think it's JSDoc style commenting (I'm not sure.) The [Docs page documentation](https://storybook.js.org/docs/react/writing-docs/docs-page) doesn't explicitly mention it, but I figured it out from the examples. I think the storybook maintainers assume that everyone comments their code like this, as a matter of course.

**Comment by shaunanoordin** [comment]
Note: I'm currently testing on `http://localhost:3000/projects/msalmon/hms-nhs-the-nautical-health-service/classify?env=production` . For some reason I'm seeing the "No workflow ID" error again, which I thought was addressed? I might have missed something. I updated the branch to master (1939972) but still no dice.

This isn't stopping me from testing this PR though (refactor + addition of text looks good), just something additional for me to check on later.

**Comment by eatyourgreens** [comment]
I thought that was fixed too. Obvious question, but have you built the latest version of the classifier component?

**Review by shaunanoordin — APPROVED** [review]
## PR Review

This PR is primarily a refactor of the Workflow Selection modal and small UI tweaks.

- Workflow Selection modal is now its own component
- Minor text updates/additions have been made

Code changes look good, and I'm noting two new changes to the coding style here.

- Components are using `/** auto-documentation comments */`, and explanations are attached to more things. I double approve! 👍 👍 
- Components don't use `MyFunctionalComponent.defaultProps()`, but the default values are instead declared in the `function MyFunctionalComponent (props)` itself.

### Testing

Tested on localhost+macOS10+Chrome88 on http://localhost:3000/projects/msalmon/hms-nhs-the-nautical-health-service/classify?env=production

- components render correctly, text updates appear where they should
- ❔ There is the odd question of the returning "Reference Error: No Workflow ID", but that error's extant even on `master` so it's not related to this PR.

Everything looks fine to me.

### Status

LGTM 👍

**Pull request merged**